tools/index.ts

```diff
@@ -0,0 +1,157 @@
+/*
+  index.ts: base file for lexical model compiles
+*/
+
+// TODO: update package.json schema at api.keyman.com/schemas to cater for additional fields.
+
+/// <reference path="lexical-model.ts" />
+/// <reference path="model-info-file.ts" />
+
+import * as ts from "typescript";
+import KmpCompiler from "./kmp-compiler";
+
+let fs = require('fs');
+let path = require('path');
+
+export default class LexicalModelCompiler {
+  compile(o: LexicalModelSource) {
+    //
+    // Load the model info file
+    //
+    let model_info_file = fs.readdirSync('../').find(function(f) { return f.match(/\.model_info$/)});
+    if(!model_info_file) {
+      this.logError('Unable to find .model_info file in parent folder');
+      return false;
+    }
+    
+    let model_info: ModelInfoFile = JSON.parse(fs.readFileSync('../'+model_info_file, 'utf8'));
+
+    //
+    // Filename expectations
+    //
+    const kpsFileName = '../source/'+model_info.id+'.model.kps';
+    const kmpFileName = model_info.id+'.model.kmp';
+    const modelFileName = model_info.id+'.model.js';
+    const modelInfoFileName = model_info.id+'.model_info';
+    const sourcePath = '../source';
+    
+    //
+    // Validate the model ID
+    //
+
+    if(!model_info.id.match(/^[a-z_][a-z0-9_]*\.[a-z_][a-z0-9_]*\.[a-z_][a-z0-9_]*$/)) {
+      this.logError(
+        `The model identifier '${model_info.id}' is invalid.\n`+
+        `Must be a valid alphanumeric identifier in format (author).(bcp_47).(uniq).\n`+
+        `bcp_47 should be underscore (_) separated.`);
+      return false;
+    }
+
+    //
+    // This script is run from folder group/author/bcp47.uniq/build/ folder. We want to
+    // verify that author.bcp47.uniq is the same as the model identifier.
+    //
+
+    {
+      let p = process.cwd().split(path.sep).reverse();
+      if(p.length < 3 || p[0] != 'build' || model_info.id != p[2] + '.' + p[1]) {
+        this.logError(`Unexpected model path ${p[2]}.${p[1]}, does not match model id ${model_info.id}`);
+        return false;
+      }
+    }
+
+    //
+    // Build the compiled lexical model
+    //
+    
+    let sources: string[] = o.sources.map(function(source) { 
+      return fs.readFileSync(path.join(sourcePath, source), 'utf8'); 
+    });
+
+    let oc: LexicalModelCompiled = {id:model_info.id, format:o.format, wordBreaking:o.wordBreaking};
```

Comment:
`format` and `wordBreaking` are not listed as properties in tools/lexical-model.ts.

Reply:
`interface LexicalModelCompiled extends LexicalModel` :)